src/services/auth/auth.service.ts

```diff
@@ -130,17 +138,43 @@ export class AuthService {
     return await this.sessionRepository.findOneBySub(this.generateDidSub(did));
   }
 
-  async createEmailAndPasswordUser(
+  async #createEmailAndPasswordUser(
     email: string,
     password: string,
     user_id: string,
   ): Promise<string> {
-    await this.legacyUserRepository.createNewEmailUser({ email, user_id });
-    return await this.legacyUserAccountRepository.createNewEmailAndPasswordUser({
-      email,
-      password,
-      username: user_id,
-    });
+    try {
+      await this.legacyUserRepository.createNewEmailUser({ email, user_id });
+      return await this.legacyUserAccountRepository.createNewEmailAndPasswordUser({
+        email,
+        password,
+        username: user_id,
+      });
+    } catch (e) {
+      if (e.code == 11000) {
+        // Duplicate key error
+        throw new HttpException(
+          { reason: 'Email Password account already created!' },
+          HttpStatus.BAD_REQUEST,
+        );
+      } else {
+        console.log(e.code);
+        throw new HttpException(
+          { reason: 'Internal Server Error' },
+          HttpStatus.INTERNAL_SERVER_ERROR,
+        );
+      }
+    }
+  }
+
+  async registerEmailAndPasswordUser(email: string, password: string) {
+    const user_id = uuid();
+
+    const email_code = await this.#createEmailAndPasswordUser(email, password, user_id);
+
+    await this.emailService.sendRegistration(email, email_code);
+
+    return { access_token: this.jwtSign({ network: 'email', user_id }) };
```

Comment:
@vaultec81 so you don't want them to have an access token til they verify their email?